source/solvers/postprocessing_scalar.cc

```diff
@@ -0,0 +1,146 @@
+  inv_range_time         = 1. / total_time_for_average;
+
+  // Calculate the average scalars.
+  average_scalar.equ(inv_range_time, sum_scalar_dt);
+}
+
+template <int dim>
+void
+AverageScalarInTime<dim>::initialize_vectors(
+  const IndexSet &locally_owned_dofs,
+  const IndexSet &locally_relevant_dofs,
+  const MPI_Comm &mpi_communicator)
+{
+  // Reinitialisation of the average scalars to get the right length.
+  scalar_dt.reinit(locally_owned_dofs, mpi_communicator);
+  sum_scalar_dt.reinit(locally_owned_dofs, mpi_communicator);
+  average_scalar.reinit(locally_owned_dofs, mpi_communicator);
+
+  average_scalar_with_ghost_cells.reinit(locally_owned_dofs,
+                                         locally_relevant_dofs,
+                                         mpi_communicator);
+
+  sum_scalar_dt_with_ghost_cells.reinit(locally_owned_dofs,
+                                        locally_relevant_dofs,
+                                        mpi_communicator);
+}
+
+template <int dim>
+void
+AverageScalarInTime<dim>::update_average_scalar()
+{
+  if (total_time_for_average > 1e-16)
+    {
+      inv_range_time = 1.0 / total_time_for_average;
+      // Calculate the average scalar.
+      average_scalar.equ(inv_range_time, sum_scalar_dt);
+    }
+}
+
+template <int dim>
+void
+AverageScalarInTime<dim>::prepare_for_mesh_adaptation()
+{
+  sum_scalar_dt_with_ghost_cells = sum_scalar_dt;
+  solution_transfer_sum_scalar_dt.prepare_for_coarsening_and_refinement(
+    sum_scalar_dt_with_ghost_cells);
+}
+
+template <int dim>
+void
+AverageScalarInTime<dim>::post_mesh_adaptation()
+{
+  solution_transfer_sum_scalar_dt.interpolate(sum_scalar_dt);
+
+  sum_scalar_dt_with_ghost_cells = sum_scalar_dt;
+
+  update_average_scalar();
+}
+
+template <int dim>
+std::vector<const GlobalVectorType *>
+AverageScalarInTime<dim>::save(const std::string &prefix)
+{
+  sum_scalar_dt_with_ghost_cells = sum_scalar_dt;
+
+  std::vector<const GlobalVectorType *> avg_scalar_set_transfer;
+  avg_scalar_set_transfer.push_back(&sum_scalar_dt_with_ghost_cells);
+
+  std::string   filename = prefix + ".averagescalar";
+  std::ofstream output(filename.c_str());
+  output << "Average scalar" << std::endl;
+  output << "dt_0 " << dt_0 << std::endl;
+  output << "Average_calculation_boolean " << average_calculation << std::endl;
+  output << "Real_initial_time " << real_initial_time << std::endl;
+
+  return avg_scalar_set_transfer;
+}
+
+template <int dim>
+std::vector<GlobalVectorType *>
+AverageScalarInTime<dim>::read(const std::string &prefix)
+{
+  std::vector<GlobalVectorType *> sum_vectors;
+  sum_vectors.push_back(&sum_scalar_dt_with_ghost_cells);
+
+  std::string   filename = prefix + ".averagescalar";
+  std::ifstream input(filename.c_str());
+  AssertThrow(input, ExcFileNotOpen(filename));
+
+  std::string buffer;
+  std::getline(input, buffer);
+  input >> buffer >> dt_0;
+  input >> buffer >> average_calculation;
+  input >> buffer >> real_initial_time;
+
```

Comment:
We could use boost serialize for this but since we don't do it for the average velocity, what I would is open an issue, leave this PR as-is, and eventually migrate veerything to boost serialize. Although @lpsaavedra and I we had some issue with boost recently for unknown reasons.